internal/validator/handler.go

```diff
@@ -42,9 +44,57 @@ type handler struct {
 	producer *kafka.Producer
 	schema   *jsonschema.Schema
 	errors   chan error
+	msgChan  chan *kafka.Message
+	vChan    chan messageInfo
 }
 
 func (this *handler) onMessage(ctx context.Context, msg *kafka.Message) {
+	wg.Add(1)
```

Comment:
Also, you may want to move some of the initial cpu-bound operations (deserialization of the message, context building) here so that the prefetch worker (i.e. handleRequest) can be focused purely on I/O

Reply:
Makes sense! Moved some of the operations here. Let me know if we should move a little more :+1: